accounts/queries.py

```diff
@@ -114,39 +114,37 @@ def get_child_participation_eligibility(child, study) -> bool:
     Returns:
         bool: Return true if child is eligible based on their prior study participation
     """
-    # Need to import StudyType here due to circular import problems
-    from studies.models import StudyType
 
-    must_have_participated = True
-    must_not_have_participated = True
+    ember_frame_player_id = 1
+    must_have = True
+    must_not = True
 
     # for both must have and must not have participated, ignore responses from internal studies that are empty
+
     if study.must_have_participated.exists():
-        must_have_participated_all = [
-            i
-            for i in study.must_have_participated.all()
-            if child.responses.filter(study=i)
-            .exclude(study__study_type=StudyType.get_ember_frame_player(), sequence=[])
-            .exists()
-        ]
-
-        # only true if response exists for each of the studies in the list
-        must_have_participated = set(must_have_participated_all) == set(
-            study.must_have_participated.all()
+        must_have_count = (
+            child.responses.filter(study__in=study.must_have_participated.all())
+            .exclude(study__study_type_id=ember_frame_player_id, sequence=[])
+            .distinct()
+            .values_list("study")
+            .count()
```

Comment:
Are we totally sure that this works? It looks to me like it is doing this:
1) for this child, get all associated responses for all studies in the 'must have participated' list
2) exclude any responses that are from EFP studies and have an empty sequence
3) call `distinct` to remove any duplicate response rows
4) get the list of study values in this set of distinct response rows
5) count the number of studies in the list

I could be wrong, but it seems to me that the order of `distinct` and `values_list` calls should be reversed, so that we're first grabbing just the study IDs from the filtered responses, and then looking for the distinct values in that set. If `distinct` is called on the queryset with the response rows, then it will return a queryset that can contain repeated responses for the same study (as long as the response rows differ in some other way). Then we're getting the study values from that queryset, which means the list it returns can contain duplicate entries for the same study. Right?

Happy to help test if necessary, I'm just not totally sure how to set up a test case.

Reply:
Good catch, I think @becky-gilbert is right about this unless I miss something about how .distinct() is calculated!

Reply:
Thank you for explaining this offline @okaycj! I now see that the order of statements here is misleading/confusing, and the SQL query that's generated from this _IS_ selecting the studies first, before calling distinct. 

(For the record, this is totally confusing and counterintuitive to me! But I guess this is part of the behind-the-scenes ORM magic. Thanks for showing me how to use the queryset's `query` attribute to see the actual SQL query that the code generates, which I'm sure will help me figure this out in the future 👍 )

Reply:
The functions provided by the Django ORM construct the SQL query and run that query when it comes to a  QuerySet function such as `count()`.  I like to think of each function adjusting the SQL query and running it when we need to.  The first two functions add to the "WHERE" statement which filters the output.  The `distinct()` and `values_list()` update the `SELECT` statement limiting the number of columns returned.  The necessary `JOINS` are generated from the double underscores in the filter/exclude functions.  This is how they convert the imperative Python to the declarative SQL.